File: src/main/java/ua/softgroup/matrix/server/supervisor/producer/json/tracking/TrackingPeriodJson.java
package ua.softgroup.matrix.server.supervisor.producer.json.tracking;

import java.time.LocalTime;

/**
 * @author Oleksandr Tyshkovets <sg.olexander@gmail.com>
 */
public class TrackingPeriodJson {

    private LocalTime start;
    private LocalTime end;
    private String windowName;
    private String keyLogger;
    private String screenshot;

    public TrackingPeriodJson() {
    }

    public TrackingPeriodJson(LocalTime start, LocalTime end, String keyLogger, String[] screenshotAndTitle) {
        this.start = start;
        this.end = end;
        this.keyLogger = keyLogger;
        this.screenshot = screenshotAndTitle[0];
        this.windowName = screenshotAndTitle[1];
    }

    public LocalTime getStart() {
        return start;
    }

    public void setStart(LocalTime start) {
        this.start = start;
    }

    public LocalTime getEnd() {
        return end;
    }

    public void setEnd(LocalTime end) {
        this.end = end;
    }

    public String getWindowName() {
        return windowName;
    }

    public void setWindowName(String windowName) {
        this.windowName = windowName;
    }

    public String getKeyLogger() {
        return keyLogger;
    }

    public void setKeyLogger(String keyLogger) {
        this.keyLogger = keyLogger;
    }

    public String getScreenshot() {
        return screenshot;
    }

    public void setScreenshot(String screenshot) {
        this.screenshot = screenshot;
    }
}

File: src/main/scala/ua/softgroup/matrix/server/supervisor/producer/json/tracking/TrackingPeriodJson.scala
package ua.softgroup.matrix.server.supervisor.producer.json.tracking

import java.time.LocalTime

import scala.beans.BeanProperty

/**
  * @author Oleksandr Tyshkovets <olexandr.tyshkovets@gmail.com>
  */
class TrackingPeriodJson {

  @BeanProperty
  var start: LocalTime = _

  @BeanProperty
  var end: LocalTime = _

  @BeanProperty
  var windowName: String = _

  @BeanProperty
  var keyLogger: String = _

  @BeanProperty
  var screenshot: String = _

  def this(start: LocalTime, end: LocalTime, keyLogger: String, screenshotAndTitle: Array[String]) {
    this()
    this.start = start
    this.end = end
    this.keyLogger = keyLogger
    this.screenshot = screenshotAndTitle(0)
    this.windowName = screenshotAndTitle(1)
  }

  override def toString: String = {
    "TrackingPeriodJson(start=%s, end=%s, windowName=%s, keyLogger=%s, screenshot=%s)"
      .format(start, end, windowName, keyLogger, screenshot)
  }

}
